webserver/webserver/src/test/java/io/helidon/webserver/UnstableTempTest.java

```diff
@@ -0,0 +1,131 @@
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import org.junit.jupiter.api.Test;
+import org.mockito.Mockito;
+
+public class UnstableTempTest {
+
+    private static final Logger LOGGER = Logger.getLogger(UnstableTempTest.class.getName());
+
+    private static final String JAR_NAME = "test.jar";
+    private static final String FILE_NAME = "test.js";
+    private static final String FILE_CONTENT = "alert(\"Hello, World!\");";
+
+    @Test
+    void cleanedTmpDuringRuntime() throws IOException {
+        List<String> contents = new ArrayList<>(2);
+
+        Path jar = createJar();
+        URL jarUrl = new URL("jar:file:" + jar.toUri().getPath() + "!/" + FILE_NAME);
+        LOGGER.info("Generated test jar url: " + jarUrl.toString());
+        ClassPathContentHandler classPathContentHandler =
+                new ClassPathContentHandler("index.html",
+                        new ContentTypeSelector(null),
+                        "/",
+                        Thread.currentThread().getContextClassLoader());
+
+        // Empty headers
+        RequestHeaders headers = mock(RequestHeaders.class);
+        when(headers.isAccepted(any())).thenReturn(true);
+        when(headers.acceptedTypes()).thenReturn(Collections.emptyList());
+        ResponseHeaders responseHeaders = mock(ResponseHeaders.class);
+
+        ServerRequest request = Mockito.mock(ServerRequest.class);
+        Mockito.when(request.headers()).thenReturn(headers);
+        ServerResponse response = Mockito.mock(ServerResponse.class);
+        MessageBodyWriterContext ctx = MessageBodyWriterContext.create(HashParameters.create());
+        ctx.registerFilter(dataChunkPub -> {
+            String fileContent = new String(Single.create(dataChunkPub).await().bytes());
+            contents.add(fileContent);
+            return Single.just(DataChunk.create(ByteBuffer.wrap(fileContent.getBytes())));
+        });
+        Mockito.when(response.headers()).thenReturn(responseHeaders);
+        Mockito.when(response.send(Mockito.any(Function.class))).then(mock -> {
+            Function<MessageBodyWriterContext, Flow.Publisher<DataChunk>> argument = mock.getArgument(0);
+            return Single.create(argument.apply(ctx)).onError(throwable -> throwable.printStackTrace());
+        });
+
+        classPathContentHandler.sendJar(Http.Method.GET, FILE_NAME, jarUrl, request, response);
+        deleteTmpFiles();
+        classPathContentHandler.sendJar(Http.Method.GET, FILE_NAME, jarUrl, request, response);
+
+        assertThat(contents, containsInAnyOrder(FILE_CONTENT, FILE_CONTENT));
+    }
+
+    private void deleteTmpFiles() throws IOException {
+        File tempDir = File.createTempFile("tempLocator", "je").getParentFile();
+        LOGGER.info("Temp dir: " + tempDir.getAbsolutePath());
+        //Delete all temp files
+        for (File file : Objects.requireNonNull(tempDir.listFiles((dir, name) -> name.endsWith(".je")))) {
```

Comment:
This may be a big performance issue for the test if the temporary directory contains a lot of files.

Reply:
I think it would be best to allow an explicit folder to be configured for static content handler, and then send a new folder from this test as the configured value.
Then you can simply delete all files in that folder to simulate the cleanup

Reply:
Great idea thx